libs/sdk-backend-tiger/src/auth.ts

```diff
@@ -114,13 +116,26 @@ export class TigerTokenAuthProvider extends TigerAuthProviderBase {
  * -  The login page will start and drive the OIDC authentication flow. Once the flow finishes and session
  *    is set up, the login page will redirect back to the application.
  *
+ * You may use the provider's ability to use passed `NotAuthenticatedHandler` function. This will be called
+ * every time a NotAuthenticated error is raised by the backend. Your application can pass a custom handler of
+ * this event - typically something that will start driving the authentication from a single place.
+ *
+ * Note: the not authenticated handler MAY be called many times in succession so you may want to wrap it in a
+ * call guard or in a debounce.
+ *
+ * @remarks See {@link redirectToTigerAuthentication} for implementation of the NotAuthenticated handler which
+ *  you may use with this provider.
  * @public
  */
 export class ContextDeferredAuthProvider extends TigerAuthProviderBase {
-    public constructor() {
+    public constructor(private readonly notAuthenticatedHandler?: NotAuthenticatedHandler) {
```

Comment:
Should we make this accept a configuration object? Do we see this configuration expanding? If so, we sholud make this more extensible (and the optional parameter will make that harder).